src/sentry/api/endpoints/org_auth_token_details.py

```diff
@@ -0,0 +1,61 @@
+from django.utils import timezone
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import audit_log
+from sentry.api.base import control_silo_endpoint
+from sentry.api.bases.organization import OrganizationEndpoint, OrgAuthTokenPermission
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.serializers import serialize
+from sentry.models.organization import Organization
+from sentry.models.orgauthtoken import OrgAuthToken
+
+
+@control_silo_endpoint
+class OrgAuthTokenDetailsEndpoint(OrganizationEndpoint):
+    permission_classes = (OrgAuthTokenPermission,)
+
+    def get(self, request: Request, organization: Organization, token_id) -> Response:
+        try:
+            instance = OrgAuthToken.objects.get(
+                organization_id=organization.id, date_deactivated__isnull=True, id=token_id
+            )
+        except OrgAuthToken.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        return Response(serialize(instance, request.user, token=None))
+
+    def put(self, request: Request, organization, token_id):
+        try:
+            instance = OrgAuthToken.objects.get(
+                organization_id=organization.id, id=token_id, date_deactivated__isnull=True
+            )
+        except OrgAuthToken.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        name = request.data.get("name", None)
+
+        if name is not None and name != "":
```

Comment:
Not sure if we prefer to ignore this if empty or error out?

Reply:
I guess it depends, for now there is only a single field. I think I'll make it required, probably safer & clearer overall!

Reply:
I agree!